Some CI updates

ranocha opened the pull request:
Using this, CI shouldn't run anymore when changing files where it's definitely not necessary. I included `skip ci` in the commit message to check again whether it works - it doesn't. As far as I know, `skip ci` works only in `push` events and not in PRs.

sloede reviewed (approved):
LGTM.

One suggestion/question: Would it make sense to reorder the components in the name for different jobs? Right now, there's a lot of - from my point of view - secondary information that comes first, before we come to the interesting part:
![image](https://user-images.githubusercontent.com/3637659/98328620-7331ee00-1ff6-11eb-9cd9-4be9af16cfa0.png)

From personal interest I think the first item should be the "test name" (i.e., "2D" or "2D_parallel"), then the OS, and the rest is secondary (since it rarely changes or is the same anyways).

ranocha commented:
I've changed the order of the tests. Moreover, I've removed the tests on Windows and Mac OS for now, since we're just running pure Julia code. Hence, we can speed up CI - in particular, if multiple CI runs are triggered simultaneously from different PRs etc.

sloede commented:
OK. Although I'd argue that we do not just use pure Julia code but also depend on packages that depend on Jlls, eg HDF5 and MPI. Even though they *should* be tested properly on all platforms by their creators, you never know. But for now, as we are so push-heavy, I guess it would be OK to only test on Linux.

Alternatively, what about only running 2D and 2d parallel tests on all platforms? Since 2D is the most feature-rich variant, this would exercise at least some parts of Trixi elsewhere.

ranocha commented:
> Alternatively, what about only running 2D and 2d parallel tests on all platforms? Since 2D is the most feature-rich variant, this would exercise at least some parts of Trixi elsewhere.

Implemented in 88093a2.
